### DISCUSSION TITLE
What boards and BLE modules are supported by the AIOBLE library?

### DISCUSSION THREAD

**Discussion by bgoolsby** [micropython/micropython#15638] [Libraries & Drivers]
On the AIOBLE github page there is no information on which boards/modules are supported.  I see from other browsing that the ESP32 and RP2040 Pico W are supported.  I'm interested in using the Adafruit Airlift module
https://www.adafruit.com/product/4201
with an Adafruit Feather M4 Express board.  The Airlift uses an ESP32, so that looks promising, but does Micropython support a connection through the M4 processor?
Thanks,

**Comment by robert-hh** [comment]
There is a PR #12937 to add WLAN and BLE support for the SAMD51 family.

**Comment by bgoolsby** [comment]
Thanks

On Mon, Aug 12, 2024 at 2:20 AM Robert Hammelrath ***@***.***>
wrote:

> There is a PR #12937
> <https://github.com/micropython/micropython/pull/12937> to add WLAN and
> BLE support for the SAMD51 family.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/micropython/discussions/15638#discussioncomment-10310169>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHGJQT5RWQZOU5ATGCSG3ZTZRBH2TAVCNFSM6AAAAABMK7RNEKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMZRGAYTMOI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Bill Goolsby

**Reply by robert-hh** [reply]
Sorry, it was the wrong PR. It is #12464, including instructions to mandatory update the ESP32 (Airlift) firmware.
Edit: For the M4 feather board, there is a better add-on board, https://www.adafruit.com/product/4264. It just stacks onto the M4 feather board.
PR #12937 would match the Adafruit Airlift M7 board, and I attached a product 4201 to a Teensy 4.0 for testing. Besides that you can use any generic ESP32 board (no ESP32Sx, no ESP32Cx) as WiFi/BLE coprocessor.
